core/core.go

```diff
@@ -278,9 +277,7 @@ func (c *Core) updateAppendQueue() {
 	for {
 		select {
 		case <-futureTimer.C:
-			c.appendProcMu.Lock()
```

Comment:
I thought this proc lock was neccessary. Why are we dropping it?

Can't resist the pun: @gameofpointers just proc locked then dropped it

Reply:
This caused some form of Deadlock behavior, so we can revisit this once we know it works, some sort of double proc prevention is definitely useful